x-pack/plugins/siem/public/common/components/exceptions/viewer/exception_entries.test.tsx

```diff
@@ -0,0 +1,149 @@
+
+import { ExceptionEntries } from './exception_entries';
+import { getFormattedEntryMock } from '../mocks';
+
+describe('ExceptionEntries', () => {
+  test('it does NOT render the and badge if only one exception item entry exists', () => {
+    const wrapper = mount(
+      <ThemeProvider theme={() => ({ eui: euiLightVars, darkMode: false })}>
+        <ExceptionEntries
+          entries={[getFormattedEntryMock()]}
+          handleDelete={jest.fn()}
+          handleEdit={jest.fn()}
+        />
+      </ThemeProvider>
+    );
+
+    expect(wrapper.find('AndOrBadge[data-test-subj="exceptionsViewerAndBadge"]')).toHaveLength(0);
+  });
+
+  test('it renders the and badge if more than one exception item exists', () => {
+    const wrapper = mount(
+      <ThemeProvider theme={() => ({ eui: euiLightVars, darkMode: false })}>
+        <ExceptionEntries
+          entries={[getFormattedEntryMock(), getFormattedEntryMock()]}
+          handleDelete={jest.fn()}
+          handleEdit={jest.fn()}
+        />
+      </ThemeProvider>
+    );
+
+    expect(wrapper.find('AndOrBadge[data-test-subj="exceptionsViewerAndBadge"]')).toHaveLength(1);
+  });
+
+  test('it invokes "handlEdit" when edit button clicked', () => {
+    const mockHandleEdit = jest.fn();
+    const wrapper = mount(
+      <ThemeProvider theme={() => ({ eui: euiLightVars, darkMode: false })}>
+        <ExceptionEntries
+          entries={[getFormattedEntryMock()]}
+          handleDelete={jest.fn()}
+          handleEdit={mockHandleEdit}
+        />
+      </ThemeProvider>
+    );
+    const editBtn = wrapper.find('[data-test-subj="exceptionsViewerEditBtn"] button').at(0);
+    editBtn.simulate('click');
+
+    expect(mockHandleEdit).toHaveBeenCalledTimes(1);
+  });
+
+  test('it invokes "handleDelete" when delete button clicked', () => {
+    const mockHandleDelete = jest.fn();
+    const wrapper = mount(
+      <ThemeProvider theme={() => ({ eui: euiLightVars, darkMode: false })}>
+        <ExceptionEntries
+          entries={[getFormattedEntryMock()]}
+          handleDelete={mockHandleDelete}
+          handleEdit={jest.fn()}
+        />
+      </ThemeProvider>
+    );
+    const deleteBtn = wrapper.find('[data-test-subj="exceptionsViewerDeleteBtn"] button').at(0);
+    deleteBtn.simulate('click');
+
+    expect(mockHandleDelete).toHaveBeenCalledTimes(1);
+  });
+
+  test('it renders nested entry', () => {
+    const parentEntry = getFormattedEntryMock();
+    parentEntry.operator = null;
+    parentEntry.value = null;
+
+    const wrapper = mount(
+      <ThemeProvider theme={() => ({ eui: euiLightVars, darkMode: false })}>
+        <ExceptionEntries
+          entries={[parentEntry, getFormattedEntryMock(true)]}
+          handleDelete={jest.fn()}
+          handleEdit={jest.fn()}
+        />
+      </ThemeProvider>
+    );
+
+    const parentField = wrapper
+      .find('EuiTableRowCell[data-test-subj="exceptionFieldNameCell"] .euiTableCellContent')
```

Comment:
I take it these will not work with just the typical:

```ts
wrapper.find('[data-test-subj="exceptionFieldNameCell"])
```

So we can avoid the extra drilling into CSS data and limit our breakages in case CSS names change?

If so, that's fine, I have had to do it before. It just makes me sad each time I have to do it personally.

Reply:
Oversight on my part. Works without it. I went through the rest of my tests to remove similar unnecessary selectors.